src/AltinnCore/Runtime/RestControllers/ValidateController.cs

```diff
@@ -0,0 +1,317 @@
+        [Route("{org}/{app}/instances/{instanceOwnerId:int}/{instanceId:guid}/validate")]
+        public async Task<IActionResult> ValidateInstance(
+            [FromRoute] string org,
+            [FromRoute] string app,
+            [FromRoute] int instanceOwnerId,
+            [FromRoute] Guid instanceId)
+        {
+            ServiceContext serviceContext = executionService.GetServiceContext(org, app, false);
+
+            Instance instance = await instanceService.GetInstance(app, org, instanceOwnerId, instanceId);
+            if (instance == null)
+            {
+                return NotFound();
+            }
+
+            string taskId = instance.Process.CurrentTask.ElementId;
+            if (taskId == null)
+            {
+                throw new ValidationException("Unable to validate instance without a started process.");
+            }
+
+            Application application = await appService.GetApplication(org, app);
+
+            List<ValidationIssue> messages = new List<ValidationIssue>();
+            foreach (ElementType elementType in application.ElementTypes.Where(et => et.Task == taskId))
+            {
+                List<DataElement> elements = instance.Data.Where(d => d.ElementType == elementType.Id).ToList();
+
+                if (elementType.MaxCount > 0 && elementType.MaxCount < elements.Count)
+                {
+                    ValidationIssue message = new ValidationIssue
+                    {
+                        Code = ValidationIssueCodes.InstanceCodes.TooManyDataElementsOfType,
+                        Scope = "INSTANCE",
+                        Severity = ValidationIssueSeverity.Error,
+                        Description = ServiceTextHelper.GetServiceText(
+                            ValidationIssueCodes.InstanceCodes.TooManyDataElementsOfType, serviceContext.ServiceText, null, "nb-NO")
+                    };
+                    messages.Add(message);
+                }
+
+                if (elementType.MinCount > 0 && elementType.MinCount > elements.Count)
+                {
+                    ValidationIssue message = new ValidationIssue
+                    {
+                        Code = ValidationIssueCodes.InstanceCodes.TooFewDataElementsOfType,
+                        Scope = "INSTANCE",
+                        Severity = ValidationIssueSeverity.Error,
+                        Description = ServiceTextHelper.GetServiceText(
+                            ValidationIssueCodes.InstanceCodes.TooFewDataElementsOfType, serviceContext.ServiceText, null, "nb-NO")
+                    };
+                    messages.Add(message);
+                }
+
+                foreach (DataElement dataElement in elements)
+                {
+                    messages.AddRange(await ValidateDataElement(org, app, instanceOwnerId, instanceId, elementType, dataElement, serviceContext));
+                }
+            }
+
+            return Ok(messages);
```

Comment:
I am not able to see how any custom validation will be performed. I see data validation based on XSD, but validation like that requires custom code will not be evaluated.